### DISCUSSION TITLE
kableExtra changing format of whole latex pdf?

### DISCUSSION THREAD

**Discussion by mthompson89** [haozhu233/kableExtra#667] [Q&A] [answered]
Hello,
I am writing a report in rMarkdown and everything looks great except when I load the library `library(kableExtra)`.  After loading the library whether I use the functions from `kableExtra` or not it changes the format of my kable tables and moves all my figures around my pdf. 

the code before loading the `kableExtra` library looks like this:
```r
kable(SummaryTree,
    align = c(rep('c',5)),
    col.names = c("Deg","BA AGS","BA UGS","TPA AGS","TPA UGS","QMD AGS","QMD UGS"),
    caption = "Summary by Degradation Category")
```
and the table look like this
![PrettyKable](https://user-images.githubusercontent.com/80464903/138938168-ac32dfbd-4684-43b5-a000-1fa72a9c21ef.png)

after I load the `kableExtra` library the table code is:
```r
kable(SummaryTree,
    align = c(rep('c',5)),
    col.names = c("Deg","AGS","UGS","AGS","UGS","AGS","UGS"),
    caption = "Summary by Degradation Category")%>%
   kableExtra::add_header_above(c(" ","Basal Area (ft2/acre)" = 2, "Trees Per Acre" = 2, "Quadratic Mean Diameter" = 2))%>%
   kableExtra::kable_classic(latex_options = "striped", position = "center")
```

and the table looks like this
![uglyKableExtra](https://user-images.githubusercontent.com/80464903/138938211-f3a1382b-8b8b-4511-9dc1-f2b33fa07cf4.png)
<br>
but I am also curious about why it is changing the formatting of my whole document. 
without `kableExtra` it looks like this
# Overview
Paragraph of text
## study site
paragraph of text
Table 1
figure 1
## iNaturalist
paragraph of text
figure 2
## summary stats
paragraph
### trees
Table 2

After loading `kableExtra` it looks like this
# Overview
Paragraph of text
## study site
paragraph of text
figure 1
Table 1
Table 2
## iNaturalist
paragraph of text
## summary stats
paragraph
### trees
figure 2

<br>
What is going on?

**Comment by haozhu233** [comment, marked as answer]
The default one generates a markdown table and pandoc converts it to a
latex table. Loading kableExtra forcing it to be rendered as latex.

To fix it, add booktabs = T in kable and add "hold_position" to
latex_options in kable_classic

**Reply by mthompson89** [reply]
where do I add `hold_position`?

**Reply by haozhu233** [reply]
kableExtra::kable_classic(latex_options = c("striped", "hold_position"))

**Reply by mthompson89** [reply]
This is so so so close. This fixed the way my tables look and now are beautiful. but it changed the order of my figures/tables again. 
instead of being 
# Header
text
## header
text
>figure1
>table1
## header
text
>figure2
## header
text
### header
text
>table2


<b>Its</b>


# Header
text
## header
text
>figure1
>table1
## header
text
## header
text
### header
text
>figure2
>table2

**Reply by mthompson89** [reply]
I think I have figured it out. my fig2, was too big and  getting bumped where it didn't  belong.
